bin/check_pylint_diff

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+# Copyright (C) 2016 Google Inc., authors, and contributors <see AUTHORS file>
+# Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
+# Created By: miha@reciprocitylabs.com
+# Maintained By: miha@reciprocitylabs.com
+
+set -o nounset
+set -o errexit
+
+ARG1=${1:-}
+TMP_REPO=".temp_repo"
+SCRIPT=$(basename "$0")
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+
+cd "${SCRIPTPATH}/.."
+
+
+if [[ "$ARG1" == "-h" || "$ARG1" == "--help" ]]; then
+  echo "
+Usage: $SCRIPT [test_commit] [-h]
+
+This script will compare pylint error count from two different commits.
+Note: all changes that are not committed will be ignored.
+
+
+Given the commit tree:
+
+       D---E---F---G---H
+            \\         /
+             A---B---C
+
+- Running '$SCRIPT' on C will check the diff between B and C
```

Comment:
It can not dot that, since you don't know if there even is a merge base. Imagine you're on master branch with only commits A B C,  and you run it on the C commit. The script assumes you want to just check the last commit (which is the diff betwenn B and C). And in case of

```
    I---J---K---L---M
     \
       D---E---F---G---H
            \         /
             A---B---C
```

I can not just assume it should take merge-base of C-H or C-M.

Using a default branch (such as maste) is also bad since you can't be sure that the repo even contains that branch. Jenkins just pull the commits by their hashes.

So you see, doing anything special when you're on a normal commit is really hard and not worth the efford (and I'd probably do it wrong)